### DISCUSSION TITLE
Add Filtering for Relationships in Confirmation Dialogs for Manual Trigger Flows

### DISCUSSION THREAD

**Discussion by pedrofnts** [directus/directus#22945] [Feature Requests]
### Summary

I am requesting the addition of a filtering capability for relationships within confirmation dialogs in flows triggered manually in Directus. This feature would allow users to select items related to the specific item where the flow is being executed.

### Motivation

This feature would streamline workflows by allowing users to filter and select items in any type of relationship that are directly related to the item being processed. For example, I'm using a flow to partially close a service order and need to select which items from that order have been completed. However, I cannot filter the items based on the current service order (where the flow is being executed) because the filter option is not available in this context. The expected outcome is an improved user experience and increased efficiency in handling related items within flows.

### Detailed Design

Currently, in Directus, you can filter a field based on another field by using the field name in the filter, such as {{ id }}. This allows for dynamic filtering based on the value of a specific field. The proposed design would extend this capability to confirmation dialogs within manual trigger flows.

The proposed design involves enhancing the confirmation dialog within manual trigger flows to include a filtering option for any type of relationship. The filter would use the ID of the current item being processed to limit the selection to related items. The implementation would require modifications to the flow configuration schema and updates to the frontend interface to support the new filter options.

### Must Have:
- Ability to filter relationships based on the current item ID.

### Should Have:
- User interface enhancements to easily configure the filter.

### Could Have:
- Support for additional relationship types in the future.

### Won't Have:
- Breaking changes to existing flow configurations.

### Drawbacks

I do not foresee any significant drawback to implementing this feature. 

### Alternatives

Alternatives considered include manually filtering items within the user interface, but this approach is inefficient and prone to user errors. Not implementing this feature would mean continuing with less efficient workflows and potentially higher error rates in data selection.

### Adoption Strategy

This feature is not a breaking change and should be easy for existing Directus developers to adopt. No migration is necessary. The addition would enhance current projects by providing more efficient data handling capabilities.

**Comment by pepeday** [comment]
Definitely a scenario I'd be interested in! Like having a choice field filled dynamically.
